pkg/dockerregistry/server/manifestschema2handler.go

```diff
@@ -118,6 +125,14 @@ func (h *manifestSchema2Handler) Verify(ctx context.Context, skipDependencyVerif
 
 			// On error here, we always append unknown blob errors.
 			errs = append(errs, distribution.ErrManifestBlobUnknown{Digest: fsLayer.Digest})
+			continue
+		}
+		if fsLayer.Size != desc.Size {
```

Comment:
You shouldn't check it when `fsLayer.MediaType == schema2.MediaTypeForeignLayer`.